src/vcUnitConversion.h

```diff
@@ -0,0 +1,78 @@
+#ifndef vcUnitConversion_h__
+#define vcUnitConversion_h__
+
+enum vcDistanceUnit
+{
+  // Metric
+  vcDistance_Metres,
+  vcDistance_Kilometres,
+  vcDistance_Centimetres,
+  vcDistance_Millimetres,
+
+  // Imperial
+  vcDistance_USSurveyFeet,
+  vcDistance_USSurveyMiles,
+  vcDistance_USSurveyInches,
+
+  // Nautical
+  vcDistance_NauticalMiles,
+
+  vcDistance_Count
+};
```

Comment:
You should try to consider data structures to help you rather than pure logic, but you can determine what you need logically, so for each input unit you require a different output multiplier value, multiplied by the input value. This suits an N<sup>2</sup> table of multiplier values, OR a conversion to and from a standard unit for each other unit of a set... the first requires N<sup>2</sup> values but lookup is a single calculation, the standard intermediate unit style is N*2 values, but lookup can be as bad as N*2, additionally doing an extra calculation. I'd suggest as well you can add `if (source == required) return input value;`